skglm/solvers/cd_solver.py

```diff
@@ -235,6 +232,17 @@ def cd_solver(
     stop_crit = np.inf  # initialize for case n_iter=0
 
     is_sparse = sparse.issparse(X)
+
+    if fit_intercept and len(w) != n_features + 1:
+        raise ValueError(
+            "Shape of w should be %i." % (n_features + 1)
+            + " Got %i" % len(w))
+
+    if not fit_intercept and len(w) != n_features:
```

Comment:
very nice

Reply:
maybe merge with above: if len(w) != n_features + intercept ?

and add a clear message saying when fit_intercept=True w needs be of shape n_feats  + 1 (written explicitely, not as its value)